clients/unshaded/src/main/java/tachyon/client/block/RemoteBlockInStream.java

```diff
@@ -30,6 +33,13 @@
 public final class RemoteBlockInStream extends BufferedBlockInStream {
   /** The address of the worker to read the data from. */
   private final InetSocketAddress mLocation;
+  /** The returned lock id after acquiring the block lock */
+  private final Long mLockId;
+
+  /** Client to communicate with the remote worker. */
+  private final WorkerClient mWorkerClient;
```

Comment:
I think it has to be cached, creating a different client will start a new session, and we cannot do unlock because worker uses the sessionId to look up the lock. LocalBlockInStream caches the worker client likewise, and only releases it after unlocking.